src/tensors/vectors.jl

```diff
@@ -0,0 +1,23 @@
+struct OffByOneVector{T} <: AbstractVector{T}
+    data::AbstractVector{T}
+end
+
+function Base.getindex(vector::OffByOneVector{T}, index::Int) where {T}
+    return vector.data[index] + 1
+end
+
+function Base.getindex(vector::OffByOneVector{T}, index::Vararg{Int}) where {T}
+    return vector.data[index...] + 1
+end
+
+function Base.setindex!(::OffByOneVector{T}, ::Any, ::Int) where {T}
+    throw(ErrorException("`OffByOneVector` is a read only object."))
+end
+
+function Base.setindex!(::OffByOneVector{T}, ::Any, ::Vararg{Int}) where {T}
+    throw(ErrorException("`OffByOneVector` is a read only object."))
+end
+
```

Comment:
let's also implement `moveto` for this vector, it's documented in this Finch somewhere.

Reply:
Done!